From 357ac00e668f75287dde449282939e1404b1a2b6 Mon Sep 17 00:00:00 2001
From: Petr Mensik <pemensik@redhat.com>
Date: Sat, 21 Aug 2021 20:54:11 +0200
Subject: [PATCH] Support dynamic linkage of tests

If libuv is not fetched, try to find and provide libuv parameters using
pkg-config. Makes dynamic linking to shared libuv library from the
system.
---
 test/CMakeLists.txt | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index 892b4aa..1d4db6e 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -8,6 +8,7 @@ set(THREADS_PREFER_PTHREAD_FLAG ON)
 find_package(Threads REQUIRED)
 
 if(FIND_GTEST_PACKAGE)
+
     find_package(GTest REQUIRED)
 else()
     FetchContent_Declare(
@@ -44,12 +45,23 @@ endif()
 function(ADD_UVW_TEST TEST_NAME TEST_SOURCE)
     add_executable(${TEST_NAME} ${TEST_SOURCE})
 
+    if (NOT FETCH_LIBUV)
+        find_package(PkgConfig)
+        pkg_check_modules(LIBUV REQUIRED libuv)
+
+        target_include_directories(
+            ${TEST_NAME}
+            PUBLIC ${LIBUV_INCLUDE_DIRS}
+        )
+    endif(NOT FETCH_LIBUV)
+
     target_link_libraries(
         ${TEST_NAME}
         PRIVATE
             $<$<TARGET_EXISTS:uvw::uvw>:uvw::uvw>
             $<$<TARGET_EXISTS:uv::uv-static>:uv::uv-static>
             $<$<TARGET_EXISTS:uvw::uvw-static>:uvw::uvw-static>
+            ${LIBUV_LDFLAGS} ${LIBUV_LIBRARIES}
             GTest::Main
             Threads::Threads
             ${LIBRT}
-- 
2.31.1

